packages/phishing-controller/src/PhishingController.ts

```diff
@@ -566,6 +574,60 @@ export class PhishingController extends BaseController<
     }
   }
 
+  /**
+   * Scan a URL for phishing. It will only scan the hostname of the URL. It also only supports
+   * web URLs.
+   *
+   * @param url - The URL to scan.
+   * @returns The phishing detection scan result.
+   */
+  async scanUrl(url: string): Promise<PhishingDetectionScanResult> {
```

Comment:
We will probably need to register this method in an action handler, for the Snap controller to be able to call it